service/history/events/cache.go

```diff
@@ -44,6 +45,7 @@ type (
 			firstEventID int64,
 			eventID int64,
 			branchToken []byte,
+			shardID int,
```

Comment:
nit: Personal opinion, I think in most cases we put `shardID` as the first parameter. Same for other functions.